### ISSUE TITLE
Escape key on ColorPicker overview inside a drawer closes the drawer + Enter key won't work

### ISSUE THREAD

**Issue by sane6** [openvi-foundation/openvue#252]
### Describe the bug

When a color picker is inside a drawer and the color preview is opened, hitting Escape closes the drawer instead of the preview. Also, Enter does not close the preview.

### Reason for not contributing a PR

- [x] Lack of time
- [x] Unsure how to implement the fix/feature
- [x] Difficulty understanding the codebase
- [ ] Other

### Reproducer

https://stackblitz.com/edit/primevue-4-vite-issue-template-kcbrfoyp

### Environment

Chrome 135

### Vue version

3.5.13

### PrimeVue version

4.3.2

### Node version

22.14.0

### Steps to reproduce the behavior

- Click the button
- Click the color
- Hit Enter -> nothing happens
- Hit Escape and the drawer closes

### Expected behavior

Escape should close the color preview and focus the textbox. Enter and Space should validate the color and close the preview.

Copied to track in our fork; not affiliated with the original report being closed/resolved upstream

Originally reported at: [primefaces#7686](https://github.com/primefaces/primevue/issues/7686)
Originally reported by: https://github.com/cadilhac
